app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -44,6 +44,19 @@ def save_ems_infra_inventory(ems, hashes, target = nil)
       return
     end
 
+    # Check if a vm was removed and we still have disk(s)
+    if target.kind_of?(VmOrTemplate) && hashes[:vms].blank?
+      current = hashes[:vm_disks].collect { |disk| URI(disk[:storage][:ems_ref]).path.split('/').last }.uniq
+      if current.blank?
+        storage = nil
+      else
+        vm_storages = ([target.storage] + target.storages).compact.uniq
+        storage = vm_storages.select { |store| !current.include?(store.ems_ref) }
+      end
+
+      target.disconnect_storage storage
+    end
```

Comment:
I was not sure about the place where to put this change and I am open to any suggestions. The issue that I am trying to fix with this PR is that we do collect information about disks being removed or not together with a vm. In order to be able to collect this information I changed the target for vm removal event and removed vm.disconnect_storage from the flow. The code need to fetch information which in other cases is part of vm hash. Having no vm hash for this specific event forced me to created new top-level key. Based on newly created key with disks I can make a decision with what parameters to run disconnect_storage method.